libflux/flux-core/src/parser/mod.rs

```diff
@@ -370,14 +374,22 @@ impl<'input> Parser<'input> {
         if let Some(import) = imports.last() {
             end = import.base.location.end;
         }
-        let body = self.parse_statement_list();
+        let mut body = self.parse_statement_list();
         if let Some(stmt) = body.last() {
             end = stmt.base().location.end;
         }
+
+        // If no package or imports were specified, then attach
+        // the attributes to the first statement if it exists.
+        if let (None, 0, Some(stmt)) = (&pkg, imports.len(), body.first_mut()) {
```

Comment:
Why do we need attach them to the first statement? Since we directly attach them to the `File` struct can't we leave them there? In other words what happens if you just delete this condition? I would expect that the attributes then always exist on the `File` struct directly but maybe I am missing something?

Reply:
Ok reading through all the test cases I see the ambiguity. I had it backwards in my head, its not that we can't put them on the File struct rather we don't want to put them on the File struct since it would be confusing that its not on the statement. 

For example:

```
@foo
x = "foo"
```

We want `@foo` to apply to the statement not to the whole package.

What if we required a package clause in those cases?

```
package bar

@foo 
x = "foo"
```

In other words the first attributes in a file always apply to the whole file scope and if you don't want that add a package clause.

Alternatively we could do the `inner/outer` differing syntax.

Reply:
There's no reason we can't do that, but I suspect that would be more confusing since you could write what you had there:

```
@foo
x = "foo"
```

If I read this, I'm not going to be thinking about how the parser interprets things. I'm going to just say "well I put an annotation on a statement, why isn't it working?"

At the same time, it occurred to me that attributes that would be attached to a statement aren't going to correspond to ones attached to the file itself so we can just check both locations and avoid an extra syntax that would be fairly confusing in my opinion.

Another alternative is just say that `package` is required if you use the attribute in the file. In all honesty, we'll have ways to specify the edition outside of the script (like the http param or the default for the organization) so I think it might be overkill to add a new syntax specifically for the package attributes. I think rust needs the syntax more because it's possible to put a crate macro anywhere in the file and it's not restricted to the beginning of the file. I think making package required if you want to use attributes would be my favorite and I think we can find a way to make it so it's not painful through the UX by just filling it in as the default.

Reply:
Makes sense lets move forward with what you have now.